### DISCUSSION TITLE
Proposal to Ship Zero JS (Even when using Partial Hydration)

### DISCUSSION THREAD

**Discussion by pedroalonsoms** [withastro/roadmap#220] [Proposal]
I think I may have an idea on how to make astro to ship 0% JS even if using client:load.

Do you think it can be useful?

Let me know, I would appreciate any feedback.

https://medium.com/@ben10pedrin/shipping-zero-javascript-a-proposal-to-make-the-web-lightning-fast-35b2a90e5e8a

**Comment by delucis** [comment]
Thanks for the idea @ben10pedrin!

Unfortunately, I think you may have overlooked in your model that you would require JavaScript on the client to manage sending requests to the server and handling responses. Likely, that JavaScript would not be significantly more performant than clientside rendering and would introduce latency and server complexity that wouldn't be compatible with Astro's static site model. 

Glad you're thinking about this though!

**Reply by pedroalonsoms** [reply]
Oh, thanks for reading, really appreciate that!
